modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java

```diff
@@ -391,6 +394,42 @@ public long getFormInstanceId() {
 			return _ddmFormInstanceId;
 		}
 
+		String ddmStructureExternalReferenceCode = PrefsParamUtil.getString(
+			_renderRequest.getPreferences(), _renderRequest,
+			"ddmStructureExternalReferenceCode");
+
+		if (!Validator.isBlank(ddmStructureExternalReferenceCode)) {
+			ThemeDisplay themeDisplay = getThemeDisplay();
+
+			Group group = _groupLocalService.fetchGroupByExternalReferenceCode(
+				PrefsParamUtil.getString(
+					_renderRequest.getPreferences(), _renderRequest,
+					"groupExternalReferenceCode"),
+				themeDisplay.getCompanyId());
+
+			DDMStructure ddmStructure =
+				_ddmStructureLocalService.fetchStructure(
+					ddmStructureExternalReferenceCode, group.getGroupId(),
+					_portal.getClassNameId(DDMFormInstance.class));
+
+			DDMFormInstance ddmFormInstance = null;
+
+			try {
+				ddmFormInstance =
+					_ddmFormInstanceLocalService.getFormInstanceByStructureId(
+						ddmStructure.getStructureId());
```

Comment:
@p-albuquerque you don't need to declare `DDMFormInstance ddmFormInstance` outside the `try`. Make it:

```
try {
    DDMFormInstance ddmFormInstance =
}
```